### ISSUE TITLE
Add support to create a VM with a reference to an external SSH key for authentication

### ISSUE THREAD

**Issue by LaurentLesle** [Azure/terraform-azurerm-avm-res-compute-virtualmachine#18]
I want to create a linux VM where the SSH key is not generated by the module and has password authentication disabled

I am setting the following attributes (extract ...)
![image](https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/assets/4702224/f31818ca-c8a4-454c-92d5-078d2e3ad892)

Getting the following error. 

```bash
 # module.azure_virtual_machine["demo"].random_password.admin_password will be created
  + resource "random_password" "admin_password" {
      + bcrypt_hash      = (sensitive value)
      + id               = (known after apply)
      + length           = 22
      + lower            = true
      + min_lower        = 2
      + min_numeric      = 2
      + min_special      = 2
      + min_upper        = 2
      + number           = true
      + numeric          = true
      + override_special = "!#$%&()*+,-./:;<=>?@[]^_{|}~"
      + result           = (sensitive value)
      + special          = true
      + upper            = true
    }

Plan: 2 to add, 0 to change, 0 to destroy.
╷
│ Error: Missing required argument
│ 
│   with module.azure_virtual_machine["demo"].data.azurerm_key_vault_secret.admin_password[0],
│   on .terraform/modules/azure_virtual_machine/main.authentication.tf line 26, in data "azurerm_key_vault_secret" "admin_password":
│   26:   name         = var.admin_password_key_vault_secret_name
│ 
│ The argument "name" is required, but no definition was found.
╵
╷
│ Error: Missing required argument
│ 
│   with module.azure_virtual_machine["demo"].data.azurerm_key_vault_secret.admin_password[0],
│   on .terraform/modules/azure_virtual_machine/main.authentication.tf line 27, in data "azurerm_key_vault_secret" "admin_password":
│   27:   key_vault_id = var.admin_credential_key_vault_resource_id
│ 
│ The argument "key_vault_id" is required, but no definition was found.
╵
```

After analysing the main.authentication.tf it looks like the code does not support that scenario,
In that scenario, the random_password.admin_password should not be called either.

**Comment by jchancellor-ms** [comment]
Thanks for catching this @LaurentLesle.  I'm working on a fix and will try to get it merged in the next few days.

**Comment by jchancellor-ms** [comment]
Closing this issue as corrected in release v0.5.0.
